spec/unit/outputs/kafka_spec.rb

```diff
@@ -141,15 +144,16 @@
 
             # Return a custom Future that will raise an exception to simulate a Kafka send() problem.
             future = java.util.concurrent.FutureTask.new { raise org.apache.kafka.common.errors.TimeoutException.new("Failed") }
-            future.run
-            future
           else
-            m.call(*args)
+            success = args
+            future = java.util.concurrent.FutureTask.new { nil } # return no-op future
           end
+          future.tap { Thread.start { future.run } }
```

Comment:
it is not entirely clear to me why we need to run this future in another thread.

Reply:
previously all FutureTask run on the same thread in specs.
my understanding of the Kafka producer was that this isn't always the case (results are retrieved async-ly) so I though of adding some "real-world" behavior in to the mix where we get results/failures from other threads.

Reply:
giving it a few more days - otherwise I think the tests should be fine (will merge as is unless concerns are raised)